libcxx/src/filesystem/directory_iterator.cpp

```diff
@@ -235,7 +235,7 @@ recursive_directory_iterator::recursive_directory_iterator(
 }
 
 void recursive_directory_iterator::__pop(error_code* ec) {
-  _LIBCPP_ASSERT_UNCATEGORIZED(__imp_, "Popping the end iterator");
+  _LIBCPP_ASSERT_NON_NULL(__imp_, "Popping the end iterator");
```

Comment:
This instance of `_LIBCPP_ASSERT_NON_NULL` makes me wonder whether we want to have `_LIBCPP_ASSERT_NON_NULL(boolean-expr, message)` or `_LIBCPP_ASSERT_NON_NULL(ptr, message)`. All the other ones are of the form `_LIBCPP_ASSERT_NON_NULL(ptr != nullptr, msg)` but this one is `_LIBCPP_ASSERT_NON_NULL(ptr, msg)`.

If we decide that we want to keep it like all the other assertion macros and keep `_LIBCPP_ASSERT_NON_NULL(boolean-expr, msg)`, then I think I would change these few instances to `_LIBCPP_ASSERT_NON_NULL(__imp_ != nullptr, "Popping the end iterator");`.

Reply:
> I think I would change these few instances to `_LIBCPP_ASSERT_NON_NULL(__imp_ != nullptr, "Popping the end iterator");`

Done.

I think it would be great to make the assertions contain more logic rather than just accepting a boolean, but I'd rather leave this for later.